roles/lib_openshift/library/oc_adm_router.py

```diff
@@ -0,0 +1,2923 @@
+            results.append('secret' in exist_volume)
+            results.append(exist_volume['secret']['secretName'] == volume['secret']['secretName'])
+            results.append(exist_volume_mount['name'] == volume_mount['name'])
+            results.append(exist_volume_mount['mountPath'] == volume_mount['mountPath'])
+
+        elif 'emptyDir' in volume:
+            results.append(exist_volume_mount['name'] == volume['name'])
+            results.append(exist_volume_mount['mountPath'] == volume_mount['mountPath'])
+
+        elif 'persistentVolumeClaim' in volume:
+            pvc = 'persistentVolumeClaim'
+            results.append(pvc in exist_volume)
+            if results[-1]:
+                results.append(exist_volume[pvc]['claimName'] == volume[pvc]['claimName'])
+
+                if 'claimSize' in volume[pvc]:
+                    results.append(exist_volume[pvc]['claimSize'] == volume[pvc]['claimSize'])
+
+        elif 'hostpath' in volume:
+            results.append('hostPath' in exist_volume)
+            results.append(exist_volume['hostPath']['path'] == volume_mount['mountPath'])
+
+        return not all(results)
+
+    def needs_update_replicas(self, replicas):
+        ''' verify whether a replica update is needed '''
+        current_reps = self.get(DeploymentConfig.replicas_path)
+        return not current_reps == replicas
+
+# -*- -*- -*- End included fragment: lib/deploymentconfig.py -*- -*- -*-
+
+# -*- -*- -*- Begin included fragment: lib/serviceaccount.py -*- -*- -*-
+
+class ServiceAccountConfig(object):
+    '''Service account config class
+
+       This class stores the options and returns a default service account
+    '''
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, sname, namespace, kubeconfig, secrets=None, image_pull_secrets=None):
+        self.name = sname
+        self.kubeconfig = kubeconfig
+        self.namespace = namespace
+        self.secrets = secrets or []
+        self.image_pull_secrets = image_pull_secrets or []
+        self.data = {}
+        self.create_dict()
+
+    def create_dict(self):
+        ''' instantiate a properly structured volume '''
+        self.data['apiVersion'] = 'v1'
+        self.data['kind'] = 'ServiceAccount'
+        self.data['metadata'] = {}
+        self.data['metadata']['name'] = self.name
+        self.data['metadata']['namespace'] = self.namespace
+
+        self.data['secrets'] = []
+        if self.secrets:
+            for sec in self.secrets:
+                self.data['secrets'].append({"name": sec})
+
+        self.data['imagePullSecrets'] = []
+        if self.image_pull_secrets:
+            for sec in self.image_pull_secrets:
+                self.data['imagePullSecrets'].append({"name": sec})
+
+class ServiceAccount(Yedit):
+    ''' Class to wrap the oc command line tools '''
+    image_pull_secrets_path = "imagePullSecrets"
+    secrets_path = "secrets"
+
+    def __init__(self, content):
+        '''ServiceAccount cons
```

Comment:
This will fail on Python 3. See https://github.com/openshift/openshift-ansible/pull/3424